metrics/geth.go

```diff
@@ -0,0 +1,56 @@
+package metrics
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	gethmetrics "github.com/ethereum/go-ethereum/metrics"
+	gethprom "github.com/ethereum/go-ethereum/metrics/prometheus"
+
+	"cosmossdk.io/log"
+)
+
+// StartGethMetricServer starts the geth metrics server on the specified address.
+func StartGethMetricServer(ctx context.Context, log log.Logger, addr string) error {
```

Comment:
this is meant to mirror pretty closely how the API server is started in cosmos sdk here: https://github.com/cosmos/cosmos-sdk/blob/727faf04848461f248f3a84edfc73c657464cdb1/server/api/server.go#L112